ShotgunImportExport/Sources/CSVImportExport.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using UnityEngine;
+using UnityEngine.Timeline;
+using UnityEngine.Playables;
+
+public static class CsvImportExport
+{
+    static readonly int k_FPS = 24;
+    static private readonly string k_directorObjectName = "Sample Shotgun CSV import";
+    static private readonly string k_fileName = "shot.csv";
+    static private readonly string k_pathToFile = Path.Combine("Assets", "CsvImportExport", "Editor");
+    static private readonly string k_timelineObjectName = "Sample Shotgun CSV timeline";
+
+    private static Tuple<ActivationTrack, TimelineClip> CreateActivationTrackAndClip(TimelineAsset timeline, PlayableDirector director)
+    {
+            ActivationTrack track = timeline.CreateTrack<ActivationTrack>(null, "");
+            TimelineClip clip = track.CreateDefaultClip();
+            return new Tuple<ActivationTrack, TimelineClip>(track, clip);
+    }
+
+    [UnityEditor.MenuItem("Shotgun CSV/Import")]
+    public static void ImportCsv ()
+    {
+        TimelineAsset timeline;
+        PlayableDirector director;
+
+        //Create a Director object and a timeline
+        UnityEngine.GameObject director_object = new UnityEngine.GameObject();
+        director = (PlayableDirector)director_object.AddComponent(typeof(PlayableDirector));
+        director.name = k_directorObjectName;
+        timeline = ScriptableObject.CreateInstance("TimelineAsset") as TimelineAsset;
+        timeline.name = k_timelineObjectName;
+        director.playableAsset = timeline;
+        
+        foreach (var row in ReadCsvFile(Path.Combine(k_pathToFile, k_fileName)))
+        {
+            var CsvField = CreateActivationTrackAndClip(timeline, director);
+            CsvField.Item1.name = row["Shot Code"];
+            CsvField.Item2.start =  (Convert.ToDouble(row["Cut In"]) / k_FPS);
+            CsvField.Item2.duration = ((Convert.ToDouble(row["Cut Out"]) - Convert.ToDouble(row["Cut In"])) / k_FPS);
+        }
+    }
+
+    [UnityEditor.MenuItem("Shotgun CSV/Export")]
+    public static void ExportCsv()
+    {
+        GameObject director = GameObject.Find(k_directorObjectName);
+
+        // Get the actual timeline object
+        var timelineAsset = (director.GetComponent("PlayableDirector") as PlayableDirector).playableAsset as TimelineAsset;
+
+        // Each dictionary in the list represents a row in the CSV file
+        // Each value in the row is addressable by its column name
+        var toWrite = new List<Dictionary<string, string>>();
+
+        foreach (var track in timelineAsset.GetOutputTracks())
+        {
+            var dict = new Dictionary<string, string>();
+            var start = Convert.ToInt64(track.start * k_FPS);
+            var end = Convert.ToInt64((track.duration + track.start) * k_FPS);

```

Comment:
Why specify time in frames rather than seconds?

Reply:
Because Shotgun uses frames